File: wup1/JS/Makes10.js
/*
   makes10

   Given 2 ints, a and b, return true if one if them is 10 or if their sum is 10.
 */

function makes10(a, b){
   if (a+b == 10 || (a == 10 | b ==10))
      return true;
   
   return false;
}

const assert = require('assert');
assert.equal(makes10(9, 10), true)
assert.equal(makes10(9, 9), false)
assert.equal(makes10(1, 9), true) 
assert.equal(makes10(10, 1), true)
assert.equal(makes10(10, 10), true)	
assert.equal(makes10(8, 2), true)
assert.equal(makes10(8, 3), false)	
assert.equal(makes10(10, 42), true)
assert.equal(makes10(12, -2), true)

console.log("All tests passed");

File: wup1/Python/Makes10.py
'''
   makes10

   Given 2 ints, a and b, return true if one if them is 10 or if their sum is 10.

'''

def makes10(a, b):
   if ((a + b == 10) or (a == 10 or b == 10)):
      return True
   return False

def tests():
   assert (makes10(9, 10) == True)
   assert (makes10(9, 9) == False)
   assert (makes10(1, 9) == True) 
   assert (makes10(10, 1) == True)
   assert (makes10(10, 10) == True)	
   assert (makes10(8, 2) == True)
   assert (makes10(8, 3) == False)	
   assert (makes10(10, 42) == True)
   assert (makes10(12, -2) == True)
   print("All tests passed")
    
tests()